src/components/pages/LoginPage/index.tsx

```diff
@@ -0,0 +1,119 @@
+import { FC, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { Form, Field } from "react-final-form";
+import { Navigate } from "react-router-dom";
+
+import Button from "../../atoms/Button";
+import ErrorMessage from "../../atoms/ErrorMessage";
+import useAuth from "../../../auth/AuthProvider";
+import * as S from "./styles";
+import formInputsValidation from "./validations";
+
+const LoginPage: FC = () => {
+  const { t } = useTranslation("", { keyPrefix: "loginPage" });
+  const [passwordShown, setPasswordShown] = useState(false);
+  const [checked, setChecked] = useState(false);
+  const { signin, user, error } = useAuth();
+
+  const handleCheckboxChange = () => {
+    setChecked(!checked);
+  };
+
+  const onSubmit = (values: { email: string; password: string }) => {
+    signin(values.email, values.password);
+  };
+
+  const togglePassword = () => {
+    setPasswordShown(!passwordShown);
+  };
+
+  return !user ? (
+    <S.LoginPageWrapper>
+      <S.LeftTopBubblesImg />
+      <S.LeftBottomBubblesImg />
+      <S.RightBubblesImg />
+      <S.LoginFormWrapper>
+        <S.H4>{t("loginHeader")}</S.H4>
+        <S.InfoText>{t("loginSubtitle")}</S.InfoText>
+        <S.StyledLink to="/">{t("learnMore")}</S.StyledLink>
+        {error && (
+          <S.LoginErrorMessage>{t("loginErrorMessage")}</S.LoginErrorMessage>
+        )}
+        <Form
+          onSubmit={onSubmit}
+          validate={formInputsValidation}
+          render={({ handleSubmit, submitting }) => (
+            <S.LoginForm onSubmit={handleSubmit}>
+              <Field name="email">
+                {({ input, meta }) => (
+                  <S.InputWrapper>
+                    <S.Label>{t("email")}</S.Label>
+                    <S.Input
+                      type="text"
+                      {...input}
+                      placeholder={t("inputEmailPlaceholder")}
+                      validationFailed={meta.touched && meta.error}
+                    />
+                    {meta.touched && meta.error && (
+                      <ErrorMessage>
+                        <S.AlertIcon />
+                        {meta.error}
+                      </ErrorMessage>
+                    )}
+                  </S.InputWrapper>
+                )}
+              </Field>
+              <Field name="password">
+                {({ input, meta }) => (
+                  <S.InputWrapper>
+                    <S.Label>
+                      {t("password")}
+                      <S.ShownPasswordButton
+                        onClick={togglePassword}
+                        aria-pressed="false"
+                        aria-label={t("showPasswordIconAlt")}
+                        role="switch"
+                      >
+                        {passwordShown ? <S.EyeIcon /> : <S.EyeOffIcon />}
```

Comment:
When I'm clicking on eyeIcon and don't have any value inputted then i got an error "Nieprawidłowy login lub hasło", this is strange🤔

Reply:
Weird.. we'll look at this 👩‍🎓

Reply:
Fixed :)